### ISSUE TITLE
Zotero integration: all users should be allowed to refresh Zotero (group) library imports

### ISSUE THREAD

**Issue by mdingemanse** [overleaf/overleaf#1065]
Zotero group libraries are shared libraries that many teams use in collaboration. It's great if Overleaf supports them. Right now it doesn't really — only the user who originally added the group library is allowed to refresh it, which defeats the point of using group libraries in the first place.

## Steps to Reproduce
1. User 1 adds a Zotero _group library_ through Add Files > From Zotero
2. User 2 clicks on zotero.bib, hits 'Refresh' button 
3. "Access Denied: You are not the user who originally imported this file"

## Expected Behaviour

Expected behaviour is that at step #2, Overleaf dutifully refreshes the bib file because it is a group library.

## Observed Behaviour

What actually happens is an access denied error: "Access Denied: You are not the user who originally imported this file"

## Context

Group libraries are shared libraries that multiple users contribute to and that Zotero syncs from multiple user accounts. For Overleaf to restrict refreshing them to just the user who happened to add it defeats the point of the group library. 

## Technical Info

The Zotero API is designed to enable this kind of use, so Overleaf is here imposing an arbitrary limitation. The issue is made for group libraries (for this is the use case that is most strongly affected), but one might argue that given the collaborative nature of Overleaf, refreshing any imported Zotero file (whether it is a personal or a group library) should be allowed by any user with rights to edit that document. After all refreshing is a harmless one-way import.

**Comment by cswpy** [comment]
This should not be a hard fix? Experiencing the same issue, not a big issue but a bit annoying.

**Comment by maugavilla** [comment]
Is there an update on the progress of this fix? Our research team has found the same issue

**Comment by JuliusHenke** [comment]
I am also experiencing this issue with my research group and am looking forward to a fix.

**Comment by mdingemanse** [comment]
Hey Overleaf @aeaton-overleaf @tomh-overleaf @alex-overleaf sorry for mass-tagging but since it's been a year and the issue persists, and bites us with any new paper we work on, I'm wondering whether you might be able to point us in the right direction for a fix. We would love to create PR ourselves if we can to fix this, or at least receive guidance on whether a fix might be on the cards somehow.

**Comment by dmackinnon1** [comment]
We tell users to work around this by not using the integration feature at all, and instead use a public group and a URL that everyone can refresh without any problems. I think you can do it with a private group as well if you generate a token and put it in the url, but I've never tried it. Instructions that we provide for users are here: https://app.frontapp.com/open/cnv_bk9ti2p?key=_reHr58eExGk7eowWwWQzJ0Fbxq8kdIZ.

**Comment by JuliusHenke** [comment]
The workaround that we are currently using is that each author creates their own Zotero references file, which they can then refresh:
- references_author1.bib
- references_author2.bib

We then include all references files in the main tex document:
```latex
\bibliography{references_author1, references_author2}
```

**Comment by mdingemanse** [comment]
Interesting. Doesn't that result in a lot of duplicates that latex will complain about? As we are all using the same group library those bib files would be literally identical when synced

**Comment by JuliusHenke** [comment]
> Interesting. Doesn't that result in a lot of duplicates that latex will complain about? As we are all using the same group library those bib files would be literally identical when synced

No, we also use the same Zotero group library. We do not get duplicates or warnings when using pdfLaTeX and BibLaTeX.

**Comment by abviv** [comment]
Let me bring this again for this year. Our project is also demanding this.

**Comment by z4lem** [comment]
Same issue here, and still no response from Overleaf.
Group libs of Zotero are a crucial feature for us, so please add the full support, i.e., including sync/access rights for all group members.

**Comment by mdingemanse** [comment]
It's been nearly three years. Thanks @das7pad for tagging this as an enhancement back in Sept 2024 (the first and so far only sign that someone has taken notice) — but really the current implementation is a bug, and we're not asking for a new feature, just for making the existing feature usable.

**Comment by chrisyeh96** [comment]
+1

I just started using Overleaf's integration with Zotero, and not being able to have collaborators refresh Zotero group library imports has been rather frustrating.

@das7pad

**Comment by dmackinnon1** [comment]
Just to note: This limitation can be avoided if, instead of using the "import bib" from Zotero option, you use the "advanced reference search" to build your in-project bib dynamically from Zotero. It's a different model for working with your bib file, so you may find that workflow has other trade-offs.

Also, issues in this repository are intended for the open-source community edition. For questions or feedback on the cloud service at overleaf.com, please get in touch with Overleaf support: https://www.overleaf.com/contact.
